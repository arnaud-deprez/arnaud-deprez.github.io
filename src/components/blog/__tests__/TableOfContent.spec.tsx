import React from 'react'
import { render } from '@testing-library/react'
import { TableOfContent } from '../.'

describe('<TableOfContent/>', () => {
  it('renders the table of content', () => {
    const input = [
      {
        value: 'heading 2a',
        depth: 2
      },
      {
        value: 'heading 2b',
        depth: 2
      },
      {
        value: 'heading 2b3a',
        depth: 3
      },
      {
        value: 'heading 2b3b',
        depth: 3
      },
      {
        value: 'heading 2b3a4a',
        depth: 4
      },
      {
        value: 'heading 2c',
        depth: 2
      }
    ]

    const { asFragment } = render(<TableOfContent headings={input} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
