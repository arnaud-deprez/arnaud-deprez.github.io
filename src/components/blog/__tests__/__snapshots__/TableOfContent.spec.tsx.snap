// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableOfContent/> renders the table of content 1`] = `
<DocumentFragment>
  <nav
    aria-label="Text navigation"
    class="table-content"
  >
    <mark>
      Table of content...
    </mark>
    <ul>
      <li>
        <a
          href="#heading 2a"
        >
          heading 2a
        </a>
      </li>
      <li>
        <a
          href="#heading 2b"
        >
          heading 2b
        </a>
      </li>
      <ul>
        <li>
          <a
            href="#heading 2b3a"
          >
            heading 2b3a
          </a>
        </li>
        <li>
          <a
            href="#heading 2b3b"
          >
            heading 2b3b
          </a>
        </li>
        <ul>
          <li>
            <a
              href="#heading 2b3a4a"
            >
              heading 2b3a4a
            </a>
          </li>
        </ul>
      </ul>
      <li>
        <a
          href="#heading 2c"
        >
          heading 2c
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
